src/Platforms/PostgreSQLPlatform.php

```diff
@@ -833,6 +833,12 @@ public function getDropIndexSQL($index, $table = null)
             return $this->getDropConstraintSQL($constraintName, $table);
         }
 
+        $tableName = $table instanceof Table ? $table->getQuotedName($this) : $table;
+        if ($tableName !== null && strpos($tableName, '.') !== false) {
+            [$schema] = explode('.', $tableName);
+            $index    = $schema . '.' . ($index instanceof Index ? $index->getQuotedName($this) : $index);
```

Comment:
This duplicates the already existing `$index instanceof Index` condition. Instead, let's initialize `$indexName` in the beginning of the method and then use it.

Reply:
There is an `$index instanceof Index` condition above, but that is paired with other conditions (primary inex) and has an early return. I think it can reach this point with both types.

Reply:
The current version looks fine. Thanks for the rework.